README.md

```diff
@@ -107,3 +107,10 @@ M.merge.top([1, 2, 3, 4]); // returns [[1, 2, 3, 4], [5, 6, 7, 8]]
 M = matrix([1, 2, 3 ,4]);
 M.merge.bottom([5, 6, 7, 8]); // returns [[1, 2, 3, 4], [5, 6, 7, 8]]
 ```
+
+#### 17. Map
+Applies a given function over the matrix, elementwise. Similar to Array.map()
+```javascript
+M = matrix([1, 2, 3]);
+M.map(x => x*x); // returns [1, 4, 9]
```

Comment:
Wouldn't `M.mul(M)` help? Were you going for solving broader use cases?

Reply:
That's correct, going for broader use.  x * x seemed like a very easy to read test function.

By the way, there's not much in the way of validation and type checking in the library, so I left off checking the return type of the function.  Should I throw an error if the return type is non-numerical?

Reply:
I'll add a second example in the readme, showing the arguments provided.

Reply:
@zxol , the entire module is intended to support numerical operations. Since, the current functions doing numerical operations don't have it, we don't need to do type checking and throw an error. We can leave the results as `NaN`.

Reply:
gotcha. 

p.s. This is my first PR.  Thanks for your time and explanations! ^^

Reply:
Welcome to the collaborative world of Github. 😃

Reply:
Thank you! Thanks for approving my PR! :smile:

Reply:
You are welcome.